src/components/themed/TransactionListComponents.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+
+import * as React from 'react'
+import { ActivityIndicator, View } from 'react-native'
+import { connect } from 'react-redux'
+
+import { type RootState } from '../../types/reduxTypes.js'
+import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext.js'
+import { EdgeText } from '../themed/EdgeText.js'
+import { TransactionListHeader } from '../themed/TransactionListHeader.js'
+
+function HeaderComponent(props: { loading: boolean }) {
+  return props.loading ? <ActivityIndicator style={{ flex: 1, alignSelf: 'center' }} size="large" /> : <TransactionListHeader />
+}
+
+export function SectionHeader(props: { title?: string }) {
+  const styles = getStyles(useTheme())
+  return (
+    <View style={styles.headerContainer}>
+      <EdgeText style={styles.headerDate}>{props.title || ''}</EdgeText>
+    </View>
+  )
+}
+
+// Spacer only
+export function SectionFooter() {
+  const theme = useTheme()
+  return <View style={{ height: theme.rem(0.5) }} />
+}
+
+const getStyles = cacheStyles((theme: Theme) => ({
+  headerContainer: {
+    marginBottom: theme.rem(0.125),
+    backgroundColor: theme.tileBackground,
+    padding: theme.rem(0.25)
+  },
+  headerDate: {
+    marginVertical: theme.rem(0.25),
+    fontSize: theme.rem(0.75),
+    color: theme.secondaryText
+  }
+}))
+
+export const Header = connect((state: RootState): { loading: boolean } => ({ loading: !state.ui.wallets.byId[state.ui.wallets.selectedWalletId] }))(
```

Comment:
Pass in the wallet ID and do not assume you can use the selected Wallet ID

Reply:
Fixed